ldms/python/ldmsd/ldmsd_communicator.py

```diff
@@ -196,21 +196,29 @@
                       }
 
 def check_offset(interval_us, offset_us=None):
+    """
+    Ensure that offset provided is valid for ldmsd with given interval
+    """
     if offset_us:
         interval_us = int(interval_us)
         offset_us = int(offset_us)
         if offset_us/interval_us > .5:
             offset_us = interval_us/2
     return offset_us
 
-def fmt_status(resp):
-    status = resp['msg']
-    if resp['errcode'] == 0 and status is not None:
-        status = json.loads(resp['msg'])
-    return status
+def fmt_status(msg):
+    """
+    Format communicator status response string into json object
+    """
+    if msg is not None:
+        try:
+            msg = json.loads(msg)
+        except Exception as e:
```

Comment:
@nick-enoent  Do you want to set msg = None in this case so the caller can universally check for None to know the return value was bad? As written the msg could != None, but is otherwise text garbage in which case the caller gets a garbage string returned instead of either None or a valid a JSON object.